### PR TITLE
Raw repository documentation 

### PR DESCRIPTION
**Pull request by mosabua**
Refactored and updated the old chapter about Maven sites and adjusted it to the raw repo concept.

https://issues.sonatype.org/browse/NEXUS-8386

### CODE REVIEW

**Comment by mosabua** [comment]
CI build at http://bamboo.s/browse/BOOK-NEXUS23-2

**Comment by mosabua** [comment]
Rendered results at http://bamboo.s/browse/BOOK-NEXUS23-2/artifact/STAGE/site/3.0/reference/sites.html

**Comment by mosabua** [comment]
Updated renders from screenshot update in next build.

File: chapter-sites.asciidoc [hunk 1]
```diff
@@ -1,80 +1,62 @@
...
+== Raw Repositories, Maven Sites and More
 
-[[sites-sect-intro]]
-=== Introduction
+Nexus includes support for hosting, proxying and grouping static
+websites - the 'raw' format. Hosted repositories with this format can
+be used to store and provide a Maven-generated website. Proxy
+repositories can subsequently proxy them in other servers. The 'raw'
+format can also be used for other resources than HTML files exposed by
+straight HTTP like browsable directory structures.
```
Comment (mrprescott) [inline chapter-sites.asciidoc]: HTTP-like

File: chapter-sites.asciidoc [hunk 2]
```diff
@@ -225,65 +154,75 @@ you should see the new role in the 'Role Management' section. Click on
...
+
+A raw proxy repository can be used to proxy any static website. This
+includes a Maven site hosted in a raw repository in another
+Nexus server or a plain static website hosted on another webserver
+like Apache httpd. It can also be used to proxy directory structures
+exposed via a webserver to distribute archives such as
+`https://nodejs.org/dist/`.
+
+Grouping raw repositories is possible and can e.g., be used to
+aggregate mutliple site repositories. However keep in mind that the
```
Comment (mrprescott) [inline chapter-sites.asciidoc]: ~~mutliple~~ multiple

File: chapter-sites.asciidoc [hunk 3]
```diff
@@ -225,65 +154,75 @@ you should see the new role in the 'Role Management' section. Click on
...
+<<admin-repositories>> using the 'raw (proxy)' and the 'raw (group)'
+recipes.
+
+A raw proxy repository can be used to proxy any static website. This
+includes a Maven site hosted in a raw repository in another
+Nexus server or a plain static website hosted on another webserver
+like Apache httpd. It can also be used to proxy directory structures
+exposed via a webserver to distribute archives such as
+`https://nodejs.org/dist/`.
+
```
Comment (mrprescott) [inline chapter-sites.asciidoc]: Please note that the raw proxy repository does not do any filtering on the static assets that it caches. Any absolute URLs in cached HTML, for example, may cause clients to bypass the proxy.

File: chapter-sites.asciidoc [hunk 4]
```diff
@@ -225,65 +154,75 @@ you should see the new role in the 'Role Management' section. Click on
...
+Grouping raw repositories is possible and can e.g., be used to
+aggregate mutliple site repositories. However keep in mind that the
+raw format does not contain any logic to resolve conflicts between the
+different repositories in the group.
```
Comment (mrprescott) [inline chapter-sites.asciidoc]: I wonder if we should just explain how groups work, 'resolving conflicts' seems worrying and vague - the group repository will consult its members, in order, and use the first matching content that it finds. What's a typical example of a conflict, a directory listing?

**Comment by mrprescott** [comment]
+1 with minor comments

**Comment by joedragons** [comment]
Running as written in 6.2 returned me the below error.  Spoke with @mrprescott and we diagnosed that the repo wasn't created.  Creating a raw hosted repo named site made it work.  For people blind following along, I strongly suggest you put that there.  I followed your steps I believe literally as I imagine new folks might.

[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 2.502s
[INFO] Finished at: Wed May 27 13:46:26 EDT 2015
[INFO] Final Memory: 17M/265M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-site-plugin:3.4:deploy (default-deploy) on project sample-site: Error uploading site: File: http://localhost:8081/repository/site/./css/maven-base.css does not exist -> [Help 1]

File: chapter-sites.asciidoc [hunk 5]
```diff
@@ -225,65 +154,75 @@ you should see the new role in the 'Role Management' section. Click on
...
+http://localhost:8081/repository/site/ - Session: Disconnecting
+http://localhost:8081/repository/site/ - Session: Disconnected
+...
+[INFO] BUILD SUCCESS
+...
 ----
 
 Once the site has been published, you can load the site in a browser
-by going to http://localhost:8081/content/sites/site/[http://localhost:8081/content/sites/site/].
+by going to http://localhost:8081/repository/site/site/index.html[http://localhost:8081/repository/site/index.html].
```
Comment (joedragons) [inline chapter-sites.asciidoc]: The URL actual (not displayed) has an extra /site in it.  Manually adjusting to http://localhost:8081/repository/site/index.html works.

**Comment by joedragons** [comment]
+1 besides notes

File: chapter-sites.asciidoc [hunk 6]
```diff
@@ -108,45 +82,37 @@ separate sites hosting your javadoc and other documentation for
...
-    </extensions>
-----
-
-In addition to the +distributionManagement+ element and the +build+
-+extension+,  add the following build element that will
-configure Maven to use version 3.4 of the Maven Site plugin.
 
-.Configuring Version 3.4 of the Maven Site Plugin
-----
 <build>
```
Comment (joedragons) [inline chapter-sites.asciidoc]: Minor: All the below are nicely spaced out except for this one.  Randomly remembered when configuring something else; sorry for the note after +1=)

File: chapter-sites.asciidoc [hunk 7]
```diff
@@ -82,8 +64,8 @@ the Maven-generated project website:
 ----
 <distributionManagement>
   <site>
-    <id>nexus-site</id>
-    <url>dav:http://localhost:8081/content/sites/site/</url>
+    <id>nexus</id>
+    <url>dav:http://localhost:8081/repository/site/</url>
   </site>
 </distributionManagement>
 ----
```
Comment (joedragons) [inline chapter-sites.asciidoc]: Minor: Unlike the below "Dependency for the Maven Site Plugin for DAV Support." stuff to plug in, this was not nicely spaced out, which makes for extra work for space-retentive folks like me=)

File: chapter-sites.asciidoc (hunk 1 above)
Comment (mosabua) [inline chapter-sites.asciidoc]: Fixed

File: chapter-sites.asciidoc (hunk 2 above)
Comment (mosabua) [inline chapter-sites.asciidoc]: Fixed

File: chapter-sites.asciidoc (hunk 3 above)
Comment (mosabua) [inline chapter-sites.asciidoc]: I added a note regarding that.

File: chapter-sites.asciidoc (hunk 4 above)
Comment (mosabua) [inline chapter-sites.asciidoc]: Yeah.. I think thats better. I will change that.

File: chapter-sites.asciidoc (hunk 5 above)
Comment (mosabua) [inline chapter-sites.asciidoc]: ups.. fixed.

**Comment by mosabua** [comment]
I added a link back to the raw hosted chapter so readers dont miss it.

File: chapter-sites.asciidoc (hunk 7 above)
Comment (mosabua) [inline chapter-sites.asciidoc]: Fixed.

File: chapter-sites.asciidoc (hunk 6 above)
Comment (mosabua) [inline chapter-sites.asciidoc]: Fixed

**Pull request merged**